jr-annotation-support/src/main/java/com/fasterxml/jackson/jr/annotationsupport/AnnotationBasedValueRWModifier.java

```diff
@@ -75,4 +97,115 @@ public void writeValue(JSONWriter context, JsonGenerator g, Object value) throws
             return _valueType;
         }
     }
+
+    /**
+     * Serialize an enum using the {@link JsonValue} tagged method
+     */
+    private static class EnumJsonValueWriter implements ValueWriter {
+        private final Class<?> _valueType;
+        private final Method _jsonValueMethod;
+
+        private EnumJsonValueWriter(Class<?> _valueType, Method _jsonValueMethod) {
+            this._valueType = _valueType;
+            this._jsonValueMethod = _jsonValueMethod;
+        }
+
+        @Override
+        public void writeValue(JSONWriter context, JsonGenerator g, Object value) throws IOException {
+            try {
+                context.writeValue(_jsonValueMethod.invoke(value));
+            } catch (Exception e) {
+                throw new IOException("Cannot call JsonValue method", e);
+            }
+        }
+
+        @Override
+        public Class<?> valueType() {
+            return _valueType;
+        }
+
+        /**
+         * Scan the methods of the enum to find a {@link JsonValue} tagged method
+         * which takes no parameters and returns a {@link String}
+         * @param type the type
+         * @return either a {@link EnumJsonValueWriter} to write this enum with, or <code>null</code> if no suitable
+         * method found
+         */
+        private static EnumJsonValueWriter of(Class<?> type) {
+            if (!isEnum(type)) {
+                return null;
+            }
+            return getJsonValueFunction(type, type);
+        }
+
+        private static EnumJsonValueWriter getJsonValueFunction(Class<?> type, Class<?> inspectionType) {
+            for (Method method : inspectionType.getDeclaredMethods()) {
+                if (method.getDeclaredAnnotation(JsonValue.class) != null &&
+                        method.getParameterCount() == 0 &&
+                        method.getReturnType().equals(String.class)) {
+                    return new EnumJsonValueWriter(type, method);
+                }
+            }
+
+            Class<?> superClass = inspectionType.getSuperclass();
+            if (superClass != null) {
```

Comment:
These lead to some unnecessary work so might want to refactor so as not to check same supertypes multiple times.

Reply:
@cowtowncoder  - I can't quite see what you mean here. It should be pointed out that for enums, the hierarchy will be almost zero, so there's unlikely to be much work here. This is a recursive scan of the immediate class and its parents, with superclass, then interfaces, and it exits as early as possible. The only way I could imagine reducing work would be to add a `ConcurrentHashMap` to cache the discovered type information.

Reply:
I have removed some unnecessary rechecking of `isEnum` on the `of` methods I created - they're only called by something that _knows_ the type is an enum

Reply:
Maybe I just misread recursion code here, mostly wrt interface detection. But I think I was confused about logic and there really isn't an issue there; although it is possible that an interface may be marked implemented at multiple levels that's not a big deal (even if gets scanned more than once).
So we are good.